src/aiida/tools/dumping/processes.py

```diff
@@ -0,0 +1,448 @@
+    node_dumper.dump_yaml(process_node=process_node, output_path=output_path.resolve())
+
+    # Don't increment index for `ProcessNodes` that don't (always?) have file IO
+    # (`CalcFunctionNodes`/`WorkFunctionNodes`), such as `create_kpoints_from_distance`
+    called_links = process_node.base.links.get_outgoing(link_type=(LinkType.CALL_CALC, LinkType.CALL_WORK)).all()
+    called_links = [
+        called_link
+        for called_link in called_links
+        if not isinstance(called_link.node, (CalcFunctionNode, WorkFunctionNode))
+    ]
+
+    for index, link_triple in enumerate(sorted(called_links, key=lambda link_triple: link_triple.node.ctime), start=1):
+        child_node = link_triple.node
+        child_label = generate_node_input_label(index=index, link_triple=link_triple)
+
+        output_path_child = output_path.resolve() / child_label
+
+        # Recursive function call for `WorkChainNode``
+        if isinstance(child_node, WorkChainNode):
+            process_dumped = workchain_dump(
+                process_node=child_node,
+                output_path=output_path_child,
+                no_node_inputs=no_node_inputs,
+                use_presubmit=use_presubmit,
+                node_dumper=node_dumper,
+            )
+
+        # Dump for `CalcJobNode`
+        elif isinstance(child_node, CalcJobNode):
+            process_dumped = calcjob_dump(
+                calcjob_node=child_node,
+                output_path=output_path_child,
+                no_node_inputs=no_node_inputs,
+                use_presubmit=use_presubmit,
+                node_dumper=node_dumper,
+            )
+
+        else:
+            process_dumped = False
+
+        all_processes_dumped = all_processes_dumped and process_dumped
+
+    return all_processes_dumped
+
+
+# Separate functions for CalcJob dumping using pre_submit, as well as for the node_inputs
+def calcjob_node_inputs_dump(calcjob_node: CalcJobNode, output_path: Path, inputs_relpath: Path = Path('node_inputs')):
+    """Dump inputs of a `CalcJobNode` of type `SinglefileData` and `FolderData`.
+
+    :param calcjob_node: The `CalcJobNode` whose inputs will be dumped.
+    :type calcjob_node: CalcJobNode
+    :param output_path: The path where the inputs will be dumped.
+    :type output_path: Path
+    """
+    dump_types = (SinglefileData, FolderData)
+
+    # ? Not using the `node_class` argument of `get_incoming`, as it does not actually retrieve, e.g. a `UpfData` node
+    # ? (due to planned deprecation?)
+    # ? Instead, check for isinstance of `SinglefileData`
+    input_node_triples = calcjob_node.base.links.get_incoming(link_type=LinkType.INPUT_CALC)
+
+    for input_node_triple in input_node_triples:
+        # Select only repositories that hold objects and are of the selected dump_types
+        if len(input_node_triple.node.base.repository.list_objects()) > 0 and isinstance(
+            input_node_triple.node, dump_types
+        ):
```

Comment:
Couldn't we just keep the first part of the conditional and drop the conditional on the node type? There might be plugins out there that use data plugins that are not a subclass of `SinglefileData` or `FolderData` but still contain relevant data in their repository. Why not be generic and dump that as well?

Reply:
I agree with this now. I added the check in the past as otherwise particular files are dumped, such as `kpoints.npy` for the `kpoints` `node_input`  or `source_file` of the `create_kpoints_from_distance` call wouldn't be dumped. But on second thought, I don't think it's a problem if these files are dumped: People who don't know them from their experience with `QE` will likely ignore them anyway, and it's better not to make assumptions about the data that might be contained in the workflow.